src/Renderers/Bs4FormRenderer.php

```diff
@@ -144,13 +145,18 @@ private function controlsInit()
 			}
 
 			if ($control instanceof Controls\Button) {
-				$markAsPrimary = $control === $this->primaryButton || (!isset($this->primaryButton) && empty($usedPrimary) && $control->parent instanceof Form);
-				if ($markAsPrimary) {
+				if ($control->controlPrototype->class === null || (is_array($control->controlPrototype->class) && !Strings::contains(implode(' ', array_keys($control->controlPrototype->class)), 'btn btn-'))) {
```

Comment:
What about doing a simple check -> do not add any class when there is *something* - I mean - the "btn btn-" check is too strict.